### PR TITLE
2 update docker for app

### PR DESCRIPTION
**Pull request by surebert**
updated requirements to use torch that exists as 2.0.0+cu117 does not exist and therefore gradio app does not work

added docker-compose.yml to allow for docker compose build and docker compose up

switched docker to use the app.py version with advanced options

added example cache directory for models and output samples

upgrading to cuda 12.2 for docker

updated readme.md with updated docker instructions

added default docker cache directories to .gitignore

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,22 +11,17 @@ RUN apt-get update && \
         python3 \
         python-is-python3 \
         python3-pip \
+        python3.10-venv \
         libgl1 \
         libgl1-mesa-glx \ 
         libglib2.0-0 \
     && rm -rf /var/lib/apt/lists/*
-
+
```
Comment (yujincheng08) [inline Dockerfile:19]: Remove extra space here.

File: Dockerfile [hunk 2]
```diff
@@ -11,22 +11,17 @@ RUN apt-get update && \
...
+        python3.10-venv \
         libgl1 \
         libgl1-mesa-glx \ 
         libglib2.0-0 \
     && rm -rf /var/lib/apt/lists/*
-
+    
 ADD requirements.txt .
 
-RUN pip install torch==2.0.0+cu117 torchvision==0.15.1+cu117 torchaudio==2.0.1 --index-url https://download.pytorch.org/whl/cu117 && \
```
Comment (yujincheng08) [inline Dockerfile:21]: we may only need to switch to `pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118`. Cuda 12 requires higher version of driver, which may be a concern.

Comment (yujincheng08) [reply Dockerfile:21]: BTW we can specify pypi source from requirements.txt, so we can keep `RUN pip install -r requirements.txt` but instead update the requirements.txt with proper pypi source.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+version: "3.8"
```
Comment (yujincheng08) [inline docker-compose.yml:1]: I think it's better to write it in README.

**Comment by surebert** [comment]
@yujincheng08 Good catch on devel. I made the changes to runtime version of nvidia base model.  I was compilng stuff in side there in my testing which is why I left devel in before by accident.

I also removed whitespace and updated instructions.  

I left version 12.  It runs really fast on my 4090 and it needs a 4090 at least on the consumer side since it takes 17GB to run.  We could add instructions on how to switch to 118 and 11.8 or we could add a second Dockerfile?

image

Out of the box this version runs where the current master branches does not

**Review by surebert — COMMENTED** [review]
I think this is ready to go, tested and working

**Review by surebert — COMMENTED** [review]
added ability to easily switch between 1024, 512 and LCM version of app in docker with new env varibale APP_CONTEXT

ignoring docker cache directory from build

**Review by surebert — COMMENTED** [review]
just updating readme with docker options for switching APP_CONTEXT between 1024, 512 and LCM versions

**Review by surebert — COMMENTED** [review]
updated readme

**Review by surebert — COMMENTED** [review]
Do you need something else from me to merge this?

**Comment by lawrence-cj** [comment]
Nice work. Approved.

**Pull request merged**